azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java

```diff
@@ -113,29 +116,51 @@ private boolean needRefreshSecretKeys() {
     }
 
     private void refreshKeyVaultSecretNames() {
-        secretNames = Optional.of(keyVaultClient)
-                .map(SecretClient::listPropertiesOfSecrets)
-                .map(secretProperties -> {
-                    final List<String> secretNameList = new ArrayList<>();
-                    secretProperties.forEach(s -> {
-                        final String secretName = s.getName().replace(vaultUri + "/secrets/", "");
-                        secretNameList.add(secretName);
-                    });
-                    return secretNameList;
-                })
-                .map(Collection::stream)
-                .orElseGet(Stream::empty)
-                .map(this::toKeyVaultSecretName)
-                .distinct()
-                .collect(Collectors.toList());
+        try {
+            secretNames = Optional.of(keyVaultClient)
+                    .map(SecretClient::listPropertiesOfSecrets)
+                    .map(secretProperties -> {
+                        final List<String> secretNameList = new ArrayList<>();
+                        secretProperties.forEach(s -> {
+                            final String secretName = s.getName().replace(vaultUri + "/secrets/", "");
+                            secretNameList.add(secretName);
+                        });
+                        return secretNameList;
+                    })
+                    .map(Collection::stream)
+                    .orElseGet(Stream::empty)
+                    .map(this::toKeyVaultSecretName)
+                    .distinct()
+                    .collect(Collectors.toList());
+        } catch (RuntimeException e) {
```

Comment:
Why try to catch runtime exception here?

Reply:
The client throws these when it has an issue connecting to the key vault and then the application just crashes. When the key vault is unavailable (for whatever reason) the application should not crash.